src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/UnstructuredAddress.java

```diff
@@ -40,12 +43,24 @@ public static UnstructuredAddress fromString(final String edifact) {
             System.arraycopy(splitByColon, 1, addressLines, 0, splitByColon.length - 1);
         }
 
-        final String postCode = splitByPlus[INDEX_POSTCODE];
+        String postCode = null;
+        if (splitByPlus.length >= INDEX_POSTCODE && StringUtils.isNotBlank(splitByPlus[INDEX_POSTCODE])) {
+            postCode = splitByPlus[INDEX_POSTCODE];
+        }
```

Comment:
Note #94 is introducing `Segment#arrayGetSafe` which will let you change this to
```suggestion
        String postCode = arrayGetSafe(splitByPlus, INDEX_POSTCODE).orElse(null);
```

Reply:
Nice! Very good function!, thanks I also have changed the postCode to be final here bc318aed3964ddb1b62e1f831d9433c3f87dcc83